darwin/dataset/download_manager.py

```diff
@@ -82,9 +83,10 @@ def download_all_images_from_annotations(
         raise ValueError(f"Annotation format {annotation_format} not supported")
 
     # Verify that there is not already image in the images folder
-    unfiltered_files = images_path.rglob("*") if use_folders else images_path.glob("*")
     existing_images = {
-        image for image in unfiltered_files if is_image_extension_allowed(image.suffix)
+        image
+        for image in images_path.rglob("*")
+        if is_file_extension_allowed(image.name)
```

Comment:
We want `rglob()` in both cases, because users may download flat releases of a dataset after downloading releases with a folder structure